### DISCUSSION TITLE
Idea - Update the run object with the message.id after it is completed for better traceability

### DISCUSSION THREAD

**Discussion by jamanrique** [openai/openai-python#982] [Ideas]
Hi!

I was exploring the API and was creating my first threads and runs. After the run is finalised, I have to use `client.beta.threads.messages.list(thread_id=thread.id)` to get the list of all the messages and then check the `run_id` of the message to check the specific response. Couldn't also help to have something like`client.beta.threads.messages.list(run_id=run.id)` or `client.beta.threads.messages.list(thread_id=thread.id, run_id=run.id)`?

The way I currently understand the API is (1) Assistant (2) Threads with message (3) Run. Since a message is produced after each run, wouldn't it make sense to update the run object with the message_id after it is finalised so I can directly evaluate the message(s) of this run?

I am prototyping some things (including sending various runs) and while it is possible to link each message to a run through threads.messages.list, somehow I think this could be useful in the future.

Hope this could be useful for someone else too! Thank you for your time developing this.

**Comment by rattrayalex** [comment]
cc @logankilpatrick
